### PR TITLE
chore(data-planes): expands test coverage

### PR DESCRIPTION
**Pull request by kleinfreund**
* chore(data-planes): expands test coverage

  Expands the test coverage for data plane views.
* chore(api): parameters incorrectly being optional
* chore: removes non-standard public/private fields

### CODE REVIEW

**Comment by slonka** [comment]
please resolve conflicts :)

**Comment by kleinfreund** [comment]
> please resolve conflicts :)

@slonka Done.

File: src/app/data-planes/views/DataPlaneListView.spec.ts [hunk 1]
```diff
@@ -72,4 +81,23 @@ describe('DataPlaneListView.vue', () => {
...
+    const wrapper = await renderComponent()
+
+    const secondTableRow = wrapper.findAll('[data-testid="data-overview-table"] tbody tr')[1]
+    expect(secondTableRow.html()).toContain(dppName)
+
+    expect(secondTableRow.element.classList.contains('is-selected')).toBe(false)
+    expect(wrapper.find('[data-testid="data-plane-proxy-title"]').html()).toContain('backend')
+    expect(window.location.search.includes('name=backend'))
+
+    // The click event listener for selecting a table row is unfortunately added to table cells instead of rows so we need to trigger a click on a cell instead of the row.
```
Comment (slonka) [inline src/app/data-planes/views/DataPlaneListView.spec.ts:96]: There is no functional error here right? This is just perf / proper implem issue right? Can we create an issue for this?

Comment (kleinfreund) [reply src/app/data-planes/views/DataPlaneListView.spec.ts:96]: Indeed, it’s not a functional error. It’s not an error of any form really. The fact that the event is fired from table cells just makes it a bit annoying to replicate it in tests. I don’t think a change in the implementation is necessary.

**Pull request merged**